rubin_scheduler/scheduler/basis_functions/mask_basis_funcs.py

```diff
@@ -194,6 +195,77 @@ def _calc_value(self, conditions, indx=None):
         return result
 
 
+class AltAzShadowMaskBasisFunction(BaseBasisFunction):
+    """Mask any out of range altitudes and azimuths, then extend the
+    mask so if observations are taken in pairs, the second in the pair will
+    not have moved into a masked region.
+    """
+
+    def __init__(
+        self,
+        nside=None,
+        min_alt=20.0,
+        max_alt=82.0,
+        shadow_minutes=40.0,
+    ):
+        super(AltAzShadowMaskBasisFunction, self).__init__(nside=nside)
+        self.min_alt = np.radians(min_alt)
+        self.max_alt = np.radians(max_alt)
+        self.shadow_time = shadow_minutes / 60.0 / 24.0  # To days
+        self.result = np.zeros(hp.nside2npix(self.nside), dtype=float)
+        self.in_range = np.zeros(hp.nside2npix(self.nside), dtype=int)
+
+    def _calc_value(self, conditions, indx=None):
+        # Mask everything to start
+        result = self.result.copy() + np.nan
+
+        in_range_alt = self.in_range.copy()
+        in_range_az = self.in_range.copy()
```

Comment:
Why do these need to be copies? Is it so that the original 'result' and 'in_range' remain at 0 instead of being modified and then re-modified? 
Is it better to make the copies or is it better to just set result=np.zeros etc. each time _calc_value is called? (I think you may have tested this?).

Reply:
Yup, just resetting them to zero each time. The speed looks to be identical, so might as well call `np.zeros` and save a few lines.